pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -530,30 +535,12 @@ async def parse_content(self):
             )
             package_repodata_downloaders.append(downloader.run())
 
-        self.data.downloaders.append(package_repodata_downloaders)
-
-        # asyncio.gather is used to preserve the order of results for package repodata
-        pending = [
-            asyncio.gather(*downloaders_group) for downloaders_group in self.data.downloaders
-        ]
-        data_type_handlers = defaultdict(
-            lambda: partial(asyncio.sleep, 0),
-            {
-                self.data.package_repodata_urls["primary"]: self.parse_packages,
-                self.data.updateinfo_url: self.parse_advisories,
-            },
-        )
-
-        while pending:
-            done, pending = await asyncio.wait(pending, return_when=asyncio.FIRST_COMPLETED)
-            for downloader in done:
-                try:
-                    results = downloader.result()
-                except ClientResponseError as exc:
-                    raise HTTPNotFound(reason=_(f"File not found: {exc.request_info.url}"))
-                else:
-                    data_url = results[0].url
-                    await data_type_handlers[data_url](results)
```

Comment:
The new version is not precisely identical, the updateinfo is not downloaded in parallel with the other files, but the extra complexity to do that is absolutely not worth it.  Can be added back later if it can be done it a more simple way.

Reply:
In such case does it make sense to exclude advisory parsing from the `parse_content` and have `parse_advisories` in the list of other methods called for parsing. It looks more explicit.
All the logic related to advisories in this method can easily be moved to `parse_advisories` I think.

Up to you. No strong opinion here.

Reply:
I could be misremembering but I think there was some kind of ordering issue that made it a pain to separate out, and this was as far as I wanted to go in this PR.

You're probably right though, it can be simplified further.

Reply:
Or maybe I just thought that this was as much as I could change at once while keeping the diff readable.

Reply:
I think order matters for migration plugin case.
Here it is important for content which has relations to other content which is currently only modules.